frontend/src/cljs/imas_seamap/map/events.cljs

```diff
@@ -454,24 +455,40 @@
      :put-hash (encode-state db)
      :dispatch [:ui/hide-loading]}))
 
+(defn update-leaflet-map [db [_ leaflet-map]]
+  (when (not= leaflet-map (get-in db [:map :leaflet-map]))
+    (.on leaflet-map "zoomend"            #(re-frame/dispatch [:map/view-updated (leaflet-props %)]))
+    (.on leaflet-map "moveend"            #(re-frame/dispatch [:map/view-updated (leaflet-props %)]))
+    (.on leaflet-map "baselayerchange"    #(re-frame/dispatch [:map/base-layer-changed (.-name %)]))
+    (.on leaflet-map "click"              #(re-frame/dispatch [:map/clicked (leaflet-props %) (mouseevent->coords %)]))
+    (.on leaflet-map "popupclose"         #(re-frame/dispatch [:map/popup-closed]))
+    (.on leaflet-map "mousemove"          #(re-frame/dispatch [:ui/mouse-pos {:x (-> % .-containerPoint .-x) :y (-> % .-containerPoint .-y)}]))
+    (.on leaflet-map "mouseout"           #(re-frame/dispatch [:ui/mouse-pos nil]))
+    (.on leaflet-map "easyPrint-start"    #(re-frame/dispatch [:ui/show-loading "Preparing Image..."]))
+    (.on leaflet-map "easyPrint-finished" #(re-frame/dispatch [:ui/hide-loading]))
+
+    (assoc-in db [:map :leaflet-map] leaflet-map)))
+
+(defn update-map-view [{{:keys [leaflet-map] old-zoom :zoom old-center :center} :map} [_ {:keys [zoom center bounds instant?]}]] 
+  (when leaflet-map
+    (if instant?
```

Comment:
I was goign to chase this up -- what's the purpose of "instant?"?  It looks rather hackish at first glance.

Reply:
Ah, it's a difference of whether we want the map to fly to that point vs just instantly moving there.